### PR TITLE
Initial support for python executable

### PR DESCRIPTION
**Pull request by moonsub-kim**
#164 

I implemented to support python executable in the CRD.

To run python or jarFile, the JobSpec.JarFile is changed with optional variable
and validate that only one of them is specified.

### CODE REVIEW

File: docs/crd.md [hunk 1]
```diff
@@ -269,9 +270,11 @@ FlinkCluster
         See [more info](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/)
     - **job** (optional): Job spec. If specified, the cluster is a Flink job cluster; otherwise, it is a Flink
       session cluster.
-      - **jarFile** (required): JAR file of the job. It could be a local file or remote URI, depending on which
-        protocols (e.g., `https://`, `gs://`) are supported by the Flink image.
-      - **className** (required): Fully qualified Java class name of the job.
+      - **jarFile** (optional): JAR file of the job. It could be a local file or remote URI, depending on which
+        protocols (e.g., `https://`, `gs://`) are supported by the Flink image. `jarFile` or `python` is required
+      - **className** (optional): Fully qualified Java class name of the job.
```
Comment (moonsub-kim) [inline docs/crd.md:275]: className is optional but it is not yet updated.

https://github.com/spotify/flink-on-k8s-operator/blob/master/api/v1beta1/flinkcluster_types.go#L371

**Comment by moonsub-kim** [comment]
@regadas could you review it?
thx!

File: api/v1beta1/flinkcluster_types.go [hunk 2]
```diff
@@ -365,11 +365,14 @@ type CleanupPolicy struct {
...
 type JobSpec struct {
 	// JAR file of the job.
-	JarFile string `json:"jarFile"`
+	JarFile *string `json:"jarFile,omitempty"`
 
 	// Fully qualified Java class name of the job.
 	ClassName *string `json:"className,omitempty"`
 
+	// Python file of the job.
+	Python *string `json:"python,omitempty"`
```
Comment (regadas) [inline api/v1beta1/flinkcluster_types.go:374]: wdyt of `PythonFile` inline with `JarFile`

**Review by regadas — COMMENTED** [review]
Thank you for this @moonsub-kim; I'll have a better look at this later today/

File: api/v1beta1/flinkcluster_types.go (hunk 2 above)
Comment (moonsub-kim) [reply api/v1beta1/flinkcluster_types.go:374]: It seems good to apply your suggestion

**Review by regadas — COMMENTED** [review]
Overall the looks good! @moonsub-kim Could you add an example to the `examples` folder?

**Comment by moonsub-kim** [comment]
Due to flink does not provide an image including python, Users must be set up thier own dockerfile and upload to any docker registry, and use it in the FlinkCluster.
Instead upload my example that contains temporary Dockerfile, I will write an guide how the FlinkCluster executes to run python without beam server

In my case, the Dockerfile copied from [Docker Setup - Enabling Python](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/resource-providers/standalone/docker/#enabling-python) is uploaded into my docker-hub registry.
Then I copied `config/samples/flinkoperator_v1beta_flinkjobcluster.yaml` and changed some lines including `.spec.image.name` like below
```yaml
apiVersion: flinkoperator.k8s.io/v1beta1
kind: FlinkCluster
metadata:
  name: flinkjobcluster-sample
spec:
  flinkVersion: "1.14"
  image:
    name: gos16052/pink-squirrel:0.1.0 # customized flink image with python
  jobManager:
    accessScope: Cluster
    ports:
      ui: 8081
    resources:
      limits:
        memory: "1024Mi"
        cpu: "200m"
  taskManager:
    replicas: 2
    resources:
      limits:
        memory: "1024Mi"
        cpu: "200m"
  job:
    pythonFile: "apps/example/word_count.py" # The image contains example python file
    parallelism: 2
    restartPolicy: "Never"
  flinkProperties:
    taskmanager.numberOfTaskSlots: "1"
    taskmanager.memory.flink.size: "560mb"
    taskmanager.memory.process.size: "1024mb"
```

**Review by regadas — COMMENTED** [review]
@moonsub-kim I think it would be cool if we include support for these 2 use cases, wdyt?

```
    Run a PyFlink job which will reference Java UDF or external connectors. JAR file specified in --jarfile will be uploaded to the cluster.

$ ./bin/flink run \
      --python examples/python/table/word_count.py \
      --jarfile <jarFile>

    Run a PyFlink job with pyFiles and the main entry module specified in --pyModule:

$ ./bin/flink run \
      --pyModule table.word_count \
      --pyFiles examples/python/table
```

**Comment by moonsub-kim** [comment]
It's good to support `pyModule, pyFiles` and `python, jarfile` as arguments. I will implement it.

**Comment by eddiewang** [comment]
would this work with Beam python jobs?

**Comment by moonsub-kim** [comment]
@eddiewang
Sorry I am not sure about your comment due to I have never used Beam before.
But, this PR is only extending argument requirements to support python executable. Maybe it will work well with Apache Beam.

**Comment by moonsub-kim** [comment]
I have extended to support additional arguments and added the guide doc.
And I removed to support to pull remote file due to I'm not using `images/flink/entrypoint.sh` in the repository so I cannot test about it

**Comment by moonsub-kim** [comment]
@regadas could you review this?

**Review by regadas — APPROVED** [review]
LGTM, it will be good to have remote file support though. I think we can tackle it on separate PR.

@moonsub-kim Thanks for this.

**Pull request merged**